src/tribler/core/components/restapi/rest/utils.py

```diff
@@ -66,3 +73,42 @@ def fix_unicode_array(arr):
             new_arr.append(item)
 
     return new_arr
+
+
+def shorten(s: Optional[str], width: int = 50, placeholder='[...]', cut_at_the_end: bool = True):
+    """ Shorten a string to a given width.
+        Example: shorten('hello world', 5) -> 'hello[...]'
+    """
+    if s and len(s) > width:
+        return f'{s[:width]}{placeholder}' if cut_at_the_end else f'{placeholder}{s[len(s) - width:]}'
+    return s
+
+
+def format_frames(frame: Optional[FrameType], file_width: int = 50, value_width: int = 100) -> Iterator[str]:
+    """ Format a stack trace."""
+    while frame:
+        filename = shorten(frame.f_code.co_filename, width=file_width, cut_at_the_end=False)
```

Comment:
**nitpick:** It may be a bit easier to read the function call with the `True` argument value:
```python
filename = shorten(frame.f_code.co_filename, width=file_width, cut_at_the_beginning=True)
```
And in the reverse case don't need to specify the `False` value explicitly, as it is the default value.

But it is a tradeoff between the readability of the function signature and the actual function call.